lib/cache/repositories/impressions/redis_repository.rb

```diff
@@ -39,14 +39,15 @@ def add_bulk(key, bucketing_key, treatments_labels_change_numbers, time)
           # delete fetched impressions afterwards
           def clear
             impressions = impression_keys.each_with_object([]) do |key, memo|
-              _, _, ip, = key.split('/')
+              ip = key.split('/')[-2] # 'prefix/sdk_lang/ip/impressions.name' -> ip
```

Comment:
what happens if there is an exception parsing? can we catch, continue with next key, clear the cache anyway and perhaps log a single line per batch?